### ISSUE TITLE
[a11y]: Blog/Podcast/Press a11y cleanup

### ISSUE THREAD

**Issue by st0nebraker** [sourcegraph/about#5665]
### Audit type

General/Other

### User journey audit issue

#5503

### Problem description

1. On an XS screen (320px), the podcast index page has too much padding on the article tags. The text is not reader friendly.
2. On a S screen, the podcast posts' YouTube video overlaps and is cut off.
3. On an XS screen (320px), all social index & post pages should left-align titles to be more readable. See image below.
4. For screenreaders, overly nested `article` tags are hard to distinguish and navigate. We should limit how much we nest `article` elements. Seen on the Podcast index cards, and each social post page.

![Screen Shot 2022-08-11 at 4 55 40 PM](https://user-images.githubusercontent.com/59381432/184259598-7047d81b-c2ba-4b93-82f0-0d942d6eebc5.png)
![Screen Shot 2022-08-11 at 4 56 50 PM](https://user-images.githubusercontent.com/59381432/184259604-bf25bfb0-bb48-45a7-964f-acda5d4a8b87.png)
![Screen Shot 2022-08-11 at 5 17 06 PM](https://user-images.githubusercontent.com/59381432/184259611-cb077ab6-f456-40db-ae36-7c2b386e155a.png)
![Screen Shot 2022-08-11 at 5 12 02 PM](https://user-images.githubusercontent.com/59381432/184259620-b6eeaed2-8f77-43b9-a0f2-526cf71861cd.png)

### Expected behavior

1. There shouldn't be too much padding for text blocks on XS screens so that the content is more reader friendly.
2. Our YouTube videos should be contained in their layout on all screen sizes.
3. Text should be left aligned on XS screens to make the content more reader friendly (especially noticeable for long headers on these screen sizes).
4. We should only nest `article` tags when necessary to make for a better screen reader experience.
